problemtools/verifyproblem.py

```diff
@@ -526,22 +535,25 @@ def parse_num(s, i):
         return self._check_res
 
 
-    def run_submission(self, sub, args, timelim_low, timelim_high):
+    def run_submission(self, sub, args, timelim, timelim_low, timelim_high):
         self.info('Running on %s' % self)
-        subres1 = []
-        subres2 = []
+        subres = []
+        subres_low = []
+        subres_high = []
         on_reject = self.config['on_reject']
         for child in self._items:
             if not child.matches_filter(args.data_filter):
                 continue
-            r1, r2 = child.run_submission(sub, args, timelim_low, timelim_high)
-            subres1.append(r1)
-            subres2.append(r2)
-            if on_reject == 'break' and r2.verdict != 'AC':
+            res, res_low, res_high = child.run_submission(sub, args, timelim, timelim_low, timelim_high)
+            subres.append(res)
+            subres_low.append(res_low)
+            subres_high.append(res_high)
+            if on_reject == 'break' and res_high.verdict != 'AC':
                 break
```

Comment:
Are you saying it should continue for subres_high?

Reply:
Oops, forgot to respond to this. Theoretically what we want to emulate is running the loop
```
output = []
for time limit in lo, med, hi:
    subres = []
    for testcase:
        res = run submission with time limit
        subres.append(res)
        if on_reject == 'break' and res.verdict != 'AC':
            break
    output.append(subres)
return output
```
but with just a single submission run per test case instead of three. Letting res_high control breaking means we might not stop res_lo as soon as we theoretically should, which could confuse its grader and give another verdict than it should. In practice, this effect should be rare and it's probably not worth the code bloat to fix.

Reply:
No, that's equally broken, or actually even more so, because it could give hi an AC verdict when it shouldn't have one.

Reply:
The latest commit looks better, but still not fully right: the code ought to break once hi gets non-AC, not lo. Or do `if not active_low and not active and not active_high: break`

Reply:
Yeah I realized upon testing 😄 
I pushed a change

Reply:
Yeah, looks good to me. I don't have merge permissions though.